File: rgctl-tests/ecommerce-python/app/schemas/review.py
from datetime import datetime

from pydantic import BaseModel, Field

from app.schemas.common import ORMModel


class CreateReviewRequest(BaseModel):
    rating: int = Field(ge=1, le=5)
    comment: str = ""


class ReviewResponse(ORMModel):
    id: str
    product_id: str
    user_id: str
    rating: int
    comment: str
    created_at: datetime

File: rgctl-tests/ecommerce-rust/src/dto/review.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Deserialize)]
pub struct CreateReviewRequest {
    pub rating: i64,
    pub comment: String,
}

#[derive(Debug, Serialize)]
pub struct ReviewResponse {
    pub id: String,
    pub product_id: String,
    pub user_id: String,
    pub rating: i64,
    pub comment: String,
}
